DMGC-IPS-01/DMGC-IPS-01_ATTINY85.ino

```diff
@@ -164,25 +191,71 @@ void loop() {
         }
         delay(DEBOUNCE_DELAY);
       }
-      
+
       if(brightness_change_flag==0){
         if(color_type==COLOR_QTY-1){
           color_type=0;
+          rainbow_current=RAINBOW_START;
         }else{
           color_type = color_type+1;
         }
       }
       // Save brightness and color setting to EPROM
-      EEPROM.put(0,brightness);
-      EEPROM.put(1,color_type);
+      EEPROM.update(0,brightness);
+      EEPROM.update(1,color_type);

```

Comment:
This doesn't actually write brightness to EEPROM when color_type changes, and doesn't write color_type when brightness changes, helping preserve write endurance of the EEPROM slightly.